doc/changes/devel.rst

```diff
@@ -51,3 +51,4 @@ API changes
 ~~~~~~~~~~~
 - ``mne.preprocessing.apply_maxfilter`` and ``mne maxfilter`` have been deprecated and will be removed in 1.7. Use :func:`mne.preprocessing.maxwell_filter` (see :ref:`this tutorial <tut-artifact-sss>`) in Python or the command-line utility from MEGIN ``maxfilter`` and :func:`mne.bem.fit_sphere_to_headshape` instead (:gh:`11938` by `Eric Larson`_)
 - :func:`mne.io.kit.read_mrk` reading pickled files is deprecated using something like ``np.savetxt(fid, pts, delimiter="\t", newline="\n")`` to save your points instead (:gh:`11937` by `Eric Larson`_)
+- ``mne.io.edf.edf._read_annotations_edf`` returns an :class:`mne.Annotations` instance (:gh:`11964` by `Paul Roujansky`_)
```

Comment:
We try not to mention private modules/attributes in the changelog since it's a user-facing document, and users shouldn't need to know/care/think about or use private code. The `API changes` section in particular is about *backward-incompatible* changes that will happen soon (typically via deprecation something will be removed in the *next* released version) that they might need to adapt their code to deal with, so this probably isn't the right section. In theory a change like this should probably live in a new section about code health / maintenance, but that's a bigger issue :) Given we don't have that, putting a line in the Enhancements section saying something general about "refactoring code for mne.read_annotations" would make sense

Reply:
Makes sense! That was my first intuition, I should have asked. Thanks for the explanation 🙂